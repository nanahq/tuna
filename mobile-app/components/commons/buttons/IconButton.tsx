import {StyleProp, Text, TextStyle, TouchableOpacity, TouchableOpacityProps} from "react-native";
import {IconComponent, IconName, IconType} from "@components/commons/IconComponent";
import {tailwind} from "@tailwind";


interface IconButtonProps extends TouchableOpacityProps {
    iconName: IconName
    iconType: IconType
    iconSize: number
    iconLabel?: string
    disabled?: boolean
    textStyle?: StyleProp<TextStyle>
    iconStyle?: StyleProp<any>
}


export function IconButton (props: IconButtonProps): JSX.Element {
    const {disabled = false} = props
    return (
        <TouchableOpacity
            testID={props.testID}
            disabled={disabled}
            onPress={props.onPress}
            style={[tailwind('flex flex-col p-1'), props.style]}
        >
            {props.iconName !== undefined && props.iconType !== undefined && (
                <IconComponent
                    iconType={props.iconType}
                    size={props.iconSize}
                    name={props.iconName}
                    style={props.iconStyle}
                />
            )}

            {props.iconLabel !== undefined && (
                <Text style={props.textStyle}>
                    {props.iconLabel}
                </Text>
            )}
        </TouchableOpacity>
    )
}
