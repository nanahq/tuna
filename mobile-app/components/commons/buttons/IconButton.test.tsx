import {render} from '@testing-library/react-native'
import {IconButton} from './IconButton'

describe('<IconButton />', () => {
    it('Should match snapshot', () => {
        const rendered = render(
            <IconButton
            iconType='Feather'
            iconSize={24}
            iconName='arrow-left'
            />
        )
        // @ts-ignore
        expect(rendered.toJSON()).toMatchSnapshot()
    })
})
