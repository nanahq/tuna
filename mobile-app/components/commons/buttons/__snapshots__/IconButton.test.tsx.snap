// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "display": "flex",
      "flexDirection": "column",
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
